service/src/main/kotlin/fi/espoo/evaka/assistanceneed/vouchercoefficient/AssistanceNeedVoucherCoefficientController.kt

```diff
@@ -103,19 +119,32 @@ class AssistanceNeedVoucherCoefficientController(private val accessControl: Acce
                         Action.AssistanceNeedVoucherCoefficient.UPDATE,
                         assistanceNeedVoucherCoefficientId
                     )
+                    val existing =
+                        tx.getAssistanceNeedVoucherCoefficientById(
+                            assistanceNeedVoucherCoefficientId
+                        )
                     adjustExistingCoefficients(
                         tx,
-                        tx.getAssistanceNeedVoucherCoefficientById(
-                                assistanceNeedVoucherCoefficientId
-                            )
-                            .childId,
+                        existing.childId,
                         body.validityPeriod,
                         assistanceNeedVoucherCoefficientId
                     )
                     tx.updateAssistanceNeedVoucherCoefficient(
-                        id = assistanceNeedVoucherCoefficientId,
-                        data = body
-                    )
+                            id = assistanceNeedVoucherCoefficientId,
+                            data = body
+                        )
+                        .also {
+                            asyncJobRunner.plan(
+                                tx,
+                                listOf(
+                                    AsyncJob.GenerateFinanceDecisions.forChild(
+                                        existing.childId,
+                                        body.validityPeriod.asDateRange(),
```

Comment:
This date range should contain both the new and the old date ranges.

Reply:
Fixed